boa/src/builtins/json/mod.rs

```diff
@@ -61,12 +61,12 @@ impl Json {
     /// [spec]: https://tc39.es/ecma262/#sec-json.parse
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse
     pub(crate) fn parse(_: &Value, args: &[Value], ctx: &mut Context) -> Result<Value> {
-        match serde_json::from_str::<JSONValue>(
-            &args
-                .get(0)
-                .expect("cannot get argument for JSON.parse")
-                .to_string(ctx)?,
-        ) {
+        let arg = match args.get(0) {
+            Some(arg) => arg.to_string(ctx)?,
+            None => return ctx.throw_syntax_error("JSON.parse(): Expected argument"),
+        };
```

Comment:
```suggestion
        let arg = match args.get(0).cloned().unwrap_or_else(Value::undefined).to_string()?;
```
As @croraf said the spec does not say that a `TypeError` should be thrown when not argument is supplied.

Reply:
Thanks for the review. I updated the code to match the suggestion.

Reply:
@HalidOdat But he did throw the `SyntaxError`, so his first version was basically fine, and according to spec. But your solution is perhaps slightly more elegant.

Reply:
> @HalidOdat But he did throw the `SyntaxError`, so his first version was basically fine, and according to spec. But your solution is perhaps slightly more elegant.

Sorry, My bad, thought it was a `TypeError`